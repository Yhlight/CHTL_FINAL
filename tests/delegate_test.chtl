html {
    body {
        div {
            id: "container";
            button {
                class: "action-btn";
                text: "Button 1";
            }
            button {
                class: "action-btn";
                text: "Button 2";
            }
        }

        script {
            // This sets up the event delegation
            {{#container}}->delegate {
                target: [{{.action-btn}}],
                click: (event) => {
                    console.log(event.target.textContent + " was clicked!");
                }
            }

            // This script will add a new button after a delay,
            // proving that the delegation works for dynamic elements.
            () => {
                setTimeout(() => {
                    const newBtn = document.createElement("button");
                    newBtn.className = "action-btn";
                    newBtn.textContent = "Button 3 (Dynamic)";
                    document.getElementById("container").appendChild(newBtn);
                }, 100);
            }();
        }
    }
}
