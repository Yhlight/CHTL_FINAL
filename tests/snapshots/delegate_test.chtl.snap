<html>
  <body>
    <div id="container">
      <button class="action-btn">
        Button 1
      </button>
      <button class="action-btn">
        Button 2
      </button>
    </div>
  </body>
</html>
